seagoat/sources/chroma.py

```diff
@@ -87,28 +142,138 @@ def fetch(query_text: str, limit: int):
 
         return format_results(query_text, repository, chromadb_results)
 
-    def cache_chunk(chunk):
+    def _add_chunks_to_collection(ids: List[str], documents: List[str], metadatas: List[Dict]) -> List[int]:
```

Comment:
apart from tests, I am noticing that this part looks a little bit vibe coded. There are a lot of comments like "with better error handling". I think that such comments are redundant, because posterity would be unaware/uninterested in a previous version of the code that had worse error handling.

if it is improving upon a previous version that was committed, then a commit message would be a good place to put such messages - otherwise I think they can be deleted